PhysicsTools/PatUtils/plugins/PATJetCleanerForType1MET.cc

```diff
@@ -0,0 +1,46 @@
+#include "JetMETCorrections/Type1MET/interface/JetCleanerForType1METT.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "PhysicsTools/PatUtils/interface/PATJetCorrExtractor.h"
+
+namespace JetCleanerForType1MET_namespace
+{
+  template <>
+  class InputTypeCheckerT<pat::Jet, PATJetCorrExtractor>
+  {
+    public:
+
+     void operator()(const pat::Jet& jet) const
+     {
+       // check that pat::Jet is of PF-type
+       if ( !jet.isPFJet() )
+	 throw cms::Exception("InvalidInput")
+	   << "Input pat::Jet is not of PF-type !!\n";
+     }
+     bool isPatJet(const pat::Jet& jet) const {
+       return true;
+     }
+  };
+
+  template <>
+  class RawJetExtractorT<pat::Jet>
```

Comment:
how is this different from what's defined in  PhysicsTools/PatUtils/interface/RawJetExtractorT.h ?
Better not create unnecessary copies of the same